client/rest/src/plugins/metadata/MetadataDb.js

```diff
@@ -82,36 +81,37 @@ class MetadataDb {
 	}
 
 	async binDataByMetalId(metalId) {
-		const compositeHashes = [routeUtils.parseArgument(metal.restoreMetadataHash(metalId), 'compositeHash', 'hash256')];
-		const metadatasByCompositeHash = await this.metadatasByCompositeHash(compositeHashes);
-		const metadataEntry = metal.getMetadataEntryByCompositehash(metadatasByCompositeHash);
+		const metadatasByCompositeHash = await this.metadatasByCompositeHash([metal.extractCompositeHashFromMetalId(metalId)]);
+		if (0 === metadatasByCompositeHash.length)
+			throw Error(`could not get first chunk, it may mistake the metal ID: ${metalId}`);
+		const { metadataEntry } = metadatasByCompositeHash[0];
 		const chunks = [];
 		let counter = 1;
+		let moreData = true;
```

Comment:
rename to hasMoreData ["has" prefix is often used to indicate boolean variables]

Reply:
🙏